src/main/java/spark/http/matching/Body.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2016 - Per Wendel
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package spark.http.matching;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import spark.utils.GzipUtils;
+import spark.serialization.SerializerChain;
+
+/**
+ * Represents the 'body'
+ */
+final class Body {
+
+    private Object content;
+
+    public static Body create() {
+        return new Body();
+    }
+
+    private Body() {
+
+    }
+
+    public boolean notSet() {
+        return content == null;
+    }
+
+    public boolean isSet() {
+        return content != null;
+    }
+
+    public Object get() {
+        return content;
+    }
+
+    public void set(Object content) {
+        this.content = content;
+    }
+
+    public void serializeTo(HttpServletResponse httpResponse,
+                            SerializerChain serializerChain,
+                            HttpServletRequest httpRequest) throws IOException {
+
+        if (!httpResponse.isCommitted()) {
+            if (httpResponse.getContentType() == null) {
+                httpResponse.setContentType("text/html; charset=utf-8");
+            }
+
+            // Check if GZIP is wanted/accepted and in that case handle that
+            OutputStream responseStream = GzipUtils.checkAndWrap(httpRequest, httpResponse, true);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *responseStream*. The resource is referenced at lines 72, 74. The resource is closed at line 75. There are other execution paths that do not contain closure statements, for example, when OutputStream.flush throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).